### PR TITLE
Remodelling (model module workover)

### PR DESCRIPTION
**Pull request by andy-beer**

### CODE REVIEW

**Comment by andy-beer** [comment]
@kadijahassanali I've volunteered you for a code review of this change – let me know if you'd rather I pass it to someone else!  There is no need to go through everything line by line but it would be good if you could take a look at the structure of the new package.  The main change is splitting the old model.py module into a class full of single line wrapper methods (in _model.py) and a bunch of other modules with code grouped by functional categories.  There have also been some changes to the handling of hdf5 file access to fully support different ways of working.  I will increase test coverage in a separate branch but I'd like to get these changes into integration before writing those extra tests.

File: resqpy/model/_forestry.py [hunk 1]
```diff
@@ -0,0 +1,673 @@
...
+
+        return True
+
+    except Exception:
+
+        log.exception('(okay to continue?) failed to load part: ' + part_name)
+        return False
+
+
+def _fell_part(model, part_name):
```
Comment (kadijahassanali) [inline resqpy/model/_forestry.py:72]: great method name!

File: resqpy/model/_hdf5.py [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+        for child in rel_root:
+            if child.attrib['Id'] == 'Hdf5File' and child.attrib['TargetMode'] == 'External':
+                child.attrib['Target'] = new_hdf5_filename
+                count += 1
+    log.info(str(count) + ' hdf5 filename' + _pl(count) + ' set to: ' + new_hdf5_filename)
+    if count > 0:
+        model.set_modified()
+
+
+def _pl(i, e = False):
```
Comment (kadijahassanali) [inline resqpy/model/_hdf5.py:345]: Function is duplicated in resqpy\well\well_utils. Do we need both versions?

File: tests/test_model.py [hunk 3]
```diff
@@ -274,3 +274,108 @@ def test_model_context(tmp_path):
         # Should be empty
         crs_list = list(model4.iter_crs())
         assert len(crs_list) == 0
+
+
+def test_multiple_epc_sharing_one_hdf5(tmp_path, example_model_with_prop_ts_rels):
```
Comment (kadijahassanali) [inline tests/test_model.py:279]: this is great!

**Review by kadijahassanali — APPROVED** [review]
@andy-beer.

This looks really good. The groupings you used are logical and easy to follow and will be the blueprint I use for any further refactoring work! I heartily approve these for a merge into integration.

File: resqpy/model/_hdf5.py (hunk 2 above)
Comment (andy-beer) [reply resqpy/model/_hdf5.py:345]: I guess we should have just a single copy in some obscure olio module. It's such a tiny function I simple cut and pasted it, probably in a few places!

**Pull request merged**